### PR TITLE
Add markdown dialect attribute.

### LINKED ISSUE #724: Allow Maruku dialect for Markdown
It would be nice to be able to use the Maruku dialect from the markdown-js library instead of Gruber this provides some extra functionality such as tables and even some meta handling that would potentially allow the power to have presenter notes within markdown

### PR DESCRIPTION
**Pull request by mrf345**
Adding a markdown dialect attribute and passing it to `markdown-js` . Resolves #724

Thanks for the awesome framework!

### CODE REVIEW

File: js/impress.js [hunk 1]
```diff
@@ -1566,6 +1566,7 @@
             var markdownDivs = document.querySelectorAll( ".markdown" );
             for ( var idx = 0; idx < markdownDivs.length; idx++ ) {
               var element = markdownDivs[ idx ];
+              var dialect = element.dataset.extrasMarkdownDialect;
```
Comment (henrikingo) [inline js/impress.js]: Hi. You need to make this change in `src/impress.js` and then run `npm run build`.

File: examples/markdown/index.html [hunk 2]
```diff
@@ -120,6 +120,11 @@
   the content was written in simple Markdown.
 </div>
 
+<div class="step slide title markdown" data-rotate="-85" data-x="4000" data-y="2000" data-extras-markdown-dialect="Maruku">
+This is an example of a slide with a specific Markdown dialect.
```
Comment (henrikingo) [inline examples/markdown/index.html]: Hmm...

You have done the correct thing, but now I'm thinking maybe we should just call the attribute `data-markdown-dialect`.

File: examples/markdown/index.html [hunk 3]
```diff
@@ -120,6 +120,11 @@
...
+===============================================================
+</div>
+
```
Comment (henrikingo) [inline examples/markdown/index.html]: Can you also document the new attribute in src/plugins/README.md

File: examples/markdown/index.html (hunk 2 above)
Comment (mrf345) [reply examples/markdown/index.html]: i like the second one more, just did.

File: js/impress.js [hunk 4]
```diff
@@ -2807,22 +2808,22 @@
  *    overwritten for individual steps and media.
  *
  *    Examples:
- *    - data-media-autostart="true" data-media-autostop="true": start media on enter, stop on
+ *    - data-media-autoplay="true" data-media-autostop="true": start media on enter, stop on
```
Comment (mrf345) [inline js/impress.js]: `autostart` was changed automatically to `autoplay` after `npm run build`. I'm doing something wrong perhaps 🤔

Comment (henrikingo) [reply js/impress.js:2811]: No, it's not you. A previous commit forgot to run build.js. It is fixed in HEAD now: https://github.com/impress/impress.js/commit/99484d11b855ebd8d5f726023ca79b88e5c42d18

You can ignore this, I think it should go away in merge. Alternatively, you can do a git merge from master to get rid of it.

**Pull request merged**